### PR TITLE
Use prettier to format code

### PR DESCRIPTION
**Pull request by SimenB**
I love that project. 😄 1.4 has typescript support.
No tslint here, so it also picked up lots of inconsistencies in the ts code base.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,9 @@
   "typings": "./lib/index.d.ts",
   "scripts": {
     "lint": "eslint .",
+    "lint:fix": "eslint . --fix  && prettier --write '{src,examples}/**/*.ts' --single-quote --tab-width 4 --print-width 100 --trailing-comma all --parser typescript",
     "build": "tsc src/*.ts src/**/*.ts -d --strictNullChecks --sourceMap --outdir ./lib --lib ES6,ES5",
+    "postbuild": "prettier --write 'lib/**/*.js'",
```
Comment (SimenB) [inline package.json]: makes the output code pretty as well, where the ts compiler does weird stuff.

File: src/client.ts [hunk 2]
```diff
@@ -1,4 +1,3 @@
-'use strict';
```
Comment (SimenB) [inline src/client.ts]: since `import` is used, it's strict by default

**Comment by sveisvei** [comment]
Wonder why the coverage dropped...

**Comment by SimenB** [comment]
>Wonder why the coverage dropped...

More lines of code. Should I just lower the threshold?

**Comment by sveisvei** [comment]
@SimenB weird - e.g. storage.ts changed, but cant see there being loc changed: 

```
-----------------------------------|----------|----------|----------|----------|----------------|
File                               |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-----------------------------------|----------|----------|----------|----------|----------------|
All files                          |    98.94 |    81.58 |    92.59 |    98.84 |                |
 src                               |    98.52 |     87.5 |     87.5 |    98.44 |                |
  client.ts                        |      100 |    66.67 |      100 |      100 |                |
  data-formatter.ts                |      100 |      100 |      100 |      100 |                |
  index.ts                         |      100 |      100 |       75 |      100 |                |
  metrics.ts                       |      100 |    95.24 |      100 |      100 |                |
  repository.ts                    |      100 |    83.33 |      100 |      100 |                |
  request.ts                       |      100 |      100 |      100 |      100 |                |
  storage.ts                       |    92.31 |       75 |      100 |    92.31 |             40 |
  unleash.ts                       |    96.88 |    85.71 |       75 |    96.77 |            130 |
 src/strategy                      |      100 |       65 |      100 |      100 |                |
  application-hostname-strategy.ts |      100 |      100 |      100 |      100 |                |
  default-strategy.ts              |      100 |       50 |      100 |      100 |                |
  gradual-rollout-random.ts        |      100 |       50 |      100 |      100 |                |
  gradual-rollout-session-id.ts    |      100 |       75 |      100 |      100 |                |
  gradual-rollout-user-id.ts       |      100 |       75 |      100 |      100 |                |
  index.ts                         |      100 |      100 |      100 |      100 |                |
  remote-addresss-strategy.ts      |      100 |        0 |      100 |      100 |                |
  strategy.ts                      |      100 |       50 |      100 |      100 |                |
  user-with-id-strategy.ts         |      100 |       50 |      100 |      100 |                |
  util.ts                          |      100 |      100 |      100 |      100 |                |
-----------------------------------|----------|----------|----------|----------|----------------|
```

vs master:
```
-----------------------------------|----------|----------|----------|----------|----------------|
File                               |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
-----------------------------------|----------|----------|----------|----------|----------------|
All files                          |    99.16 |    93.09 |    96.97 |    99.53 |                |
 src                               |    98.94 |     94.7 |    95.95 |    99.42 |                |
  client.ts                        |    97.62 |    90.32 |      100 |    97.44 |             74 |
  data-formatter.ts                |      100 |      100 |      100 |      100 |                |
  index.ts                         |      100 |      100 |       75 |      100 |                |
  metrics.ts                       |      100 |      100 |      100 |      100 |                |
  repository.ts                    |      100 |      100 |      100 |      100 |                |
  request.ts                       |      100 |       75 |      100 |      100 |                |
  storage.ts                       |    97.96 |    88.89 |      100 |    97.56 |             64 |
  unleash.ts                       |    97.47 |    92.86 |     87.5 |      100 |                |
 src/strategy                      |      100 |    86.49 |      100 |      100 |                |
  application-hostname-strategy.ts |      100 |    71.43 |      100 |      100 |                |
  default-strategy.ts              |      100 |      100 |      100 |      100 |                |
  gradual-rollout-random.ts        |      100 |      100 |      100 |      100 |                |
  gradual-rollout-session-id.ts    |      100 |     87.5 |      100 |      100 |                |
  gradual-rollout-user-id.ts       |      100 |     87.5 |      100 |      100 |                |
  index.ts                         |      100 |      100 |      100 |      100 |                |
  remote-addresss-strategy.ts      |      100 |      100 |      100 |      100 |                |
  strategy.ts                      |      100 |       75 |      100 |      100 |                |
  user-with-id-strategy.ts         |      100 |      100 |      100 |      100 |                |
  util.ts                          |      100 |      100 |      100 |      100 |                |
-----------------------------------|----------|----------|----------|----------|---------------
```

**Comment by sveisvei** [comment]
But sure, you can change the threshold - but I feel its something weird going on given the uncovered stuff changed....

**Comment by SimenB** [comment]
Test coverage is run on the generated code, not the source code...

**Comment by SimenB** [comment]
So the error came from me running prettier on the built code. Removed that, and coverage remains the same

**Comment by SimenB** [comment]
>Coverage increased (+0.5%) to 100.0%

Nice

**Pull request merged**